### ISSUE TITLE
ClassCastException after opening a Twig template

### ISSUE THREAD

**Issue by sasy-lingo** [Haehnchen/idea-php-symfony2-plugin#663]
I've encountered the following error: when I open a Twig template and switch back to a PHP file within the same session, after that whenever I click in that file, I get a `ClassCastException`.

| Q | A
| --- | ---
| PhpStorm version | 10.0.2 build: PS-143.1184
| JRE | 1.8.0_51-b16 amd64
| Plugin version | 0.11.106
| Project type | Symfony fullstack
| PhpStorm EAP channel | no

This issue is _probably_ related to #592 

After I restart PhpStorm, the issue is gone until I open another Twig file.

Stacktrace:

```
com.intellij.lang.javascript.psi.impl.JSFunctionExpressionImpl cannot be cast to com.jetbrains.php.lang.psi.elements.PhpNamespace
java.lang.ClassCastException: com.intellij.lang.javascript.psi.impl.JSFunctionExpressionImpl cannot be cast to com.jetbrains.php.lang.psi.elements.PhpNamespace
	at fr.adrienbrault.idea.symfony2plugin.util.PhpIndexUtil$1.process(PhpIndexUtil.java:39)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:165)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:52)
	at com.intellij.psi.stubs.StubIndexImpl$1.process(StubIndexImpl.java:231)
	at com.intellij.psi.stubs.StubIndexImpl$StubIdListContainerAction.perform(StubIndexImpl.java:582)
	at com.intellij.psi.stubs.StubIndexImpl$StubIdListContainerAction.perform(StubIndexImpl.java:570)
	at com.intellij.util.indexing.ValueContainer.forEach(ValueContainer.java:81)
	at com.intellij.psi.stubs.StubIndexImpl.a(StubIndexImpl.java:252)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:223)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:212)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:77)
	at fr.adrienbrault.idea.symfony2plugin.util.PhpIndexUtil.getPhpClassInsideNamespace(PhpIndexUtil.java:39)
	at fr.adrienbrault.idea.symfony2plugin.util.PhpIndexUtil.getPhpClassInsideNamespace(PhpIndexUtil.java:27)
	at fr.adrienbrault.idea.symfony2plugin.doctrine.EntityHelper.getModelClasses(EntityHelper.java:789)
	at fr.adrienbrault.idea.symfony2plugin.doctrine.EntityHelper.getModelClasses(EntityHelper.java:754)
	at fr.adrienbrault.idea.symfony2plugin.config.ServiceLineMarkerProvider.entityClassMarker(ServiceLineMarkerProvider.java:151)
	at fr.adrienbrault.idea.symfony2plugin.config.ServiceLineMarkerProvider.collectSlowLineMarkers(ServiceLineMarkerProvider.java:69)
	at com.intellij.codeInsight.daemon.impl.SlowLineMarkersPass.addLineMarkers(SlowLineMarkersPass.java:78)
	at com.intellij.codeInsight.daemon.impl.SlowLineMarkersPass.doCollectInformation(SlowLineMarkersPass.java:65)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:67)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:444)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1169)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:435)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:432)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:206)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)
```

**Comment by tgabi333** [comment]
+1

**Comment by sasy-lingo** [comment]
After approx 30 mins of work:

```
com.intellij.diagnostic.MessagePool$TooManyErrorsException: Too many IDE fatal errors. Monitoring stopped.
```

:(

**Comment by sasy-lingo** [comment]
After a bit further inspection, this issue is not related to #658 
I was using `com.xenji.php.symfony2.clickableviews`, but unistalled it and the original problem still persists.

**Comment by Haehnchen** [comment]
maybe also related to #670 reported by phpstorm core dev

**Comment by Haehnchen** [comment]
fixed by #670
